### PR TITLE
New Documentation Structure

### PR DESCRIPTION
**Pull request by kotfu**
For #709. The new documentation in the PR is at least as good as what we had before, and in many ways it's better. Still a lot more work to be done on the documentation, but we can create new issues and branches for the additional work.

### CODE REVIEW

**Comment by tleonhardt** [comment]
Wow, 56 files changed!  I'll start reviewing but it might take a little while.

**Comment by tleonhardt** [comment]
I added autoclass documentation of `cmd2.Cmd` in **api/cmd.rst**.  This required whitespace tweaks to a number of docstrings within the class.

File: cmd2/cmd2.py [hunk 1]
```diff
@@ -720,23 +720,20 @@ def _reset_completion_defaults(self) -> None:
...
-                     raw_tokens: list of tokens with any quotes preserved
-                                 this can be used to know if a token was quoted or is missing a closing quote
-
-                     Both lists are guaranteed to have at least 1 item
-                     The last item in both lists is the token being tab completed
-
-                 On Failure
-                    Two empty lists
+                 **On Success**
+                 - tokens: list of unquoted tokens - this is generally the list needed for tab completion functions
```
Comment (tleonhardt) [inline cmd2/cmd2.py]: The `autodoc` extension for Sphinx is rather particular when it comes to indentation within the params/return section of docstrings.  If we want to make it look pretty in the docs, we can use full reStructuredText however.

All changes made in **cmd2.py** were made specifically to fix `autodoc` build errors.

File: docs/api/index.rst [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+=============
+
+.. toctree::
+   :maxdepth: 1
+
+   cmd
+   decorators
+   exceptions
+   utility_functions
+   utility_classes
```
Comment (tleonhardt) [inline docs/api/index.rst]: There is probably more we need to add to this, but that can be done later

File: docs/conf.py [hunk 3]
```diff
@@ -55,8 +56,8 @@
 
 # General information about the project.
 project = 'cmd2'
-copyright = '2010-2018, Catherine Devlin and Todd Leonhardt'
-author = 'Catherine Devlin and Todd Leonhardt'
+copyright = '2010-2019, Catherine Devlin and Todd Leonhardt'
```
Comment (tleonhardt) [inline docs/conf.py]: I'd recommend changing the copyright to also say "cmd2 contributors" with a link to the list of contributors on GitHub: https://github.com/python-cmd2/cmd2/graphs/contributors

File: docs/examples/first_app.rst [hunk 4]
```diff
@@ -0,0 +1,322 @@
+First Application
```
Comment (tleonhardt) [inline docs/examples/first_app.rst]: Nice job on the first application walkthrough here!

File: docs/features/index.rst [hunk 5]
```diff
@@ -0,0 +1,26 @@
+Features
+========
+
+.. toctree::
+   :maxdepth: 1
+
+   commands
```
Comment (tleonhardt) [inline docs/features/index.rst]: For a lengthy toctree like this, we should consider alphabetizing to make it easier to visually parse

**Comment by tleonhardt** [comment]
Thanks undertaking this major reorganization of our documentation structure.  This is definitely better than what we have now.  It obviously still needs a lot to be filled in, but we can tackle that over time prior to a 1.0 release.

File: docs/features/index.rst (hunk 5 above)
Comment (kotfu) [reply docs/features/index.rst]: Agreed. Alphabetic sorting, and maybe also categorizing. With a short description of each feature in the table of contents.

File: docs/conf.py (hunk 3 above)
Comment (kotfu) [reply docs/conf.py]: Done.

File: docs/features/index.rst (hunk 5 above)
Comment (tleonhardt) [reply docs/features/index.rst]: I pushed a change to make it alphabetical.  Categorizing would be useful as well.

File: docs/copyright.rst [hunk 6]
```diff
@@ -0,0 +1,7 @@
+Copyright
```
Comment (tleonhardt) [inline docs/copyright.rst]: Thanks for adding a good copyright with link to cmd2 contributors.

**Pull request merged**